docs/source/crawler/basics.rst

```diff
@@ -27,5 +27,22 @@ a modified dictionary as its output once.
    or modified.  To help with creation of a new shallow copy with some fields
    adjusted, use :func:`~datalad.utils.updated`.
 
-Pipeline
---------
\ No newline at end of file
+Pipelines
+---------
+
+A pipeline is a series of functions gathered into a list. Each function take the
+output of its predecessor as its own input. The first function in the pipeline
+would need to be provided with input. The simplest pipeline could look like
+
+>>> def pipeline():
+...     [
+...     crawl_url('http://map.org/datasets'),
+...     a_href_match(".*.mat"),
+...     annex
+...     ]
+
+which crawls in a website, which is provided as input to the first function, then
+matches to all files that end in `.mat`, and those files are lastly inputted to
+the annex functions which simply annexes them. There can also be subpiplines within
+a pipeline, which are also denoted by `[]`. Two subpipelines that exist on top of
```

Comment:
On Tue, 21 Jun 2016, Gergana Alteva wrote:

>    In docs/source/crawler/basics.rst:

>  > +
>  > +A pipeline is a series of functions gathered into a list. Each function take the
>  > +output of its predecessor as its own input. The first function in the pipeline
>  > +would need to be provided with input. The simplest pipeline could look like
>  > +
>  > +>>> def pipeline():
>  > +...     [
>  > +...     crawl_url('http://map.org/datasets'),
>  > +...     a_href_match(".*.mat"),
>  > +...     annex
>  > +...     ]
>  > +
>  > +which crawls in a website, which is provided as input to the first function, then
>  > +matches to all files that end in `.mat`, and those files are lastly inputted to
>  > +the annex functions which simply annexes them. There can also be subpiplines within
>  > +a pipeline, which are also denoted by `[]`. Two subpipelines that exist on top of

>    @yarikoptic should generators be its own section, or maybe a subsection of
>    pipelines?

yes -- I guess the "Nodes" section, which describes them could be
introduced if not there yet.

-- 
Yaroslav O. Halchenko
Center for Open Neuroscience     http://centerforopenneuroscience.org
Dartmouth College, 419 Moore Hall, Hinman Box 6207, Hanover, NH 03755
Phone: +1 (603) 646-9834                       Fax: +1 (603) 646-1419
WWW:   http://www.linkedin.com/in/yarik